scripts/cnn_variant_wdl/cram2filtered.wdl

```diff
@@ -0,0 +1,473 @@
+    }
+ 
+    call MergeBamOuts {
+        input:
+            bam_outs = RunHC4.bamout,
+            output_prefix = output_prefix,
+            ref_fasta = reference_fasta,
+            ref_dict = reference_dict,
+            ref_fai = reference_fasta_index,            
+            gatk_override = gatk_override,
+            preemptible_attempts = preemptible_attempts,
+            gatk_docker = gatk_docker                  
+    }
+
+
+    output {
+        MergeVCF_HC4.*
+        MergeBamOuts.*
+    }
+
+}
+
+
+task CramToBam {
+    File reference_fasta
+    File reference_fasta_index
+    File reference_dict
+    File cram_file
+    String output_prefix
+
+    # Runtime parameters
+    Int? mem_gb
+    Int? preemptible_attempts
+    Int? disk_space_gb
+    Int? cpu 
+    Boolean use_ssd = true
+
+    # You may have to change the following two parameter values depending on the task requirements
+    Int default_ram_mb = 16000
+    # WARNING: In the workflow, you should calculate the disk space as an input to this task (disk_space_gb).
+    Int default_disk_space_gb = 200
+
+    # Mem is in units of GB but our command and memory runtime values are in MB
+    Int machine_mem = if defined(mem_gb) then mem_gb *1000 else default_ram_mb
+    Int command_mem = machine_mem - 1000
+
+command <<<
+  ls -ltr ${cram_file} ${reference_fasta} &&
+  echo "ls (1): complete" &&
+  samtools view -h -T ${reference_fasta} ${cram_file} |
+  samtools view -b -o ${output_prefix}.bam - &&
+  echo "samtools view: complete" &&
+  ls -ltr . &&
+  echo "ls (2): complete" &&
+  samtools index -b ${output_prefix}.bam &&
+  echo "samtools index: complete" &&
+  ls -ltr . &&
+  echo "ls (3): complete" &&
+  mv ${output_prefix}.bam.bai ${output_prefix}.bai &&
+  echo "mv: complete" &&
+  ls -ltr . &&
+  echo "ls (4): complete"
+  >>>
+  runtime {
+    docker: "broadinstitute/genomes-in-the-cloud:2.1.1"
+    memory: machine_mem + " MB"
+    # Note that the space before SSD and HDD should be included.
+    disks: "local-disk " + select_first([disk_space_gb, default_disk_space_gb]) + if use_ssd then " SSD" else " HDD"
+    preemptible: select_first([preemptible_attempts, 3])
+    cpu: select_first([cpu, 1])  
+  }
+
+  output {
+    File output_bam = "${output_prefix}.bam"
+    File output_bam_index = "${output_prefix}.bai"
+  }
+}
+
+
+task RunHC4 {
+    File input_bam
+    File input_bam_index
+    File reference_fasta
+    File reference_dict
+    File reference_fasta_index
+    String output_prefix
+    File interval_list
+    String extra_args
+    File? gatk_override
+
+    # Runtime parameters
+    Int? mem_gb
+    String gatk_docker
+    Int? preemptible_attempts
+    Int? disk_space_gb
+    Int? cpu 
+    Boolean use_ssd = true
+
+    # You may have to change the following two parameter values depending on the task requirements
+    Int default_ram_mb = 8000
+    # WARNING: In the workflow, you should calculate the disk space as an input to this task (disk_space_gb).
```

Comment:
Any reason not to paste in the size calculation from https://github.com/broadinstitute/dsde-pipelines/blob/develop/genomes_in_the_cloud/single_sample/PairedSingleSampleWf.wdl ?

Reply:
Added